### ISSUE TITLE
Standard keyword $id in the schema is reported as unknown

### ISSUE THREAD

**Issue by TheNilesh** [java-json-tools/json-schema-validator#253]
[Standard keyword](http://json-schema.org/latest/json-schema-core.html#rfc.section.9.2) `$id` in the schema is reported as unknown.
```
[{
	level="warning",
	schema={
		"loadingURI": "#",
		"pointer": ""
	},
	domain="syntax",
	message="unknown keyword(s) found; ignored",
	ignored=["$id"]
}, {
  //another error
}]
```

Example Schema:
```
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "$id": "classpath:jsonschema/person.schema.json",
  "title": "Person Schema",
  "description": "Schema having $id",
  "type": "object",
  "properties": {
    "name": {
      "type": "string"
    },
    "age": {
      "type": "integer"
    }
  },
  "required": [
    "name"
  ],
  "additionalProperties": false
}
```

Sample Input JSON:
```
{
    "age":21
}
```

**Comment by rbuckland** [comment]
`$id` was introduced as a keyword to `draft-06` of the JSON-Schema Specifications.
For `draft-04`, you need to use `id`.
Reference - http://json-schema.org/draft-06/json-schema-release-notes.html
`draft-07` is the latest.

**Comment by corentinmb** [comment]
Hi

Version used of the dependency: 2.2.10

I have the same problem with a draft-07 versioned schema:
`warning: the following keywords are unknown and will be ignored: [$id, readOnly, writeOnly]`

```json
{
  "definitions": {},
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$id": "conformity-schema.json",
  "type": "object",
  "title": "...",
  "description": "...",
  "readOnly": true,
  "writeOnly": false,
...
}
```

Thanks in advance :+1:

**Comment by cricket007** [comment]
> problem with a draft-07 versioned schema

The readme seems to say that this project only supports up to v4 schemas.
